erdjs/src/smartcontracts/codec/numerical.ts

```diff
@@ -0,0 +1,129 @@
+import { NumericalType, NumericalValue } from "../typesystem";
+import { isMsbZero, isMsbOne, bigIntToBuffer, bufferToBigInt, cloneBuffer, flipBufferBitsInPlace, prependByteToBuffer } from "./utils";
+
+export class NumericalBinaryCoded {
+    /**
+     * Reads and decodes a NumericalValue from a given buffer, 
+     * with respect to: {@link https://docs.elrond.com/developers/developer-reference/the-elrond-serialization-format | The Elrond Serialization Format}. 
+     * 
+     * @param buffer the input buffer
+     * @param type the primitive type
+     */
+    decodeNested(buffer: Buffer, type: NumericalType): [NumericalValue, number] {
+        let offset = 0;
+        let length = type.sizeInBytes;
+
+        if (!length) {
+            // Size of type is not known: arbitrary-size big integer.
+            // Therefore, we must read the length from the header.
+            offset = 4;
+            length = buffer.readUInt32BE();
+        }
+
+        let payload = buffer.slice(offset, offset + length);
+        let result = this.decodeTopLevel(payload, type);
+        let decodedLength = length + offset;
+        return [result, decodedLength];
+    }
+
+    /**
+     * Reads and decodes a NumericalValue from a given buffer, 
+     * with respect to: {@link https://docs.elrond.com/developers/developer-reference/the-elrond-serialization-format | The Elrond Serialization Format}. 
+     * 
+     * @param buffer the input buffer
+     * @param type the primitive type
+     */
+    decodeTopLevel(buffer: Buffer, type: NumericalType): NumericalValue {
+        let payload = cloneBuffer(buffer);
+
+        let empty = buffer.length == 0;
+        if (empty) {
+            return new NumericalValue(BigInt(0), type);
+        }
+
+        let isNotNegative = !type.withSign || isMsbZero(payload);
```

Comment:
isPositive?

Reply:
Fixed - indeed, since 0 isn't possible at that point (handled beforehand).